test/url-parser.test.js

```diff
@@ -0,0 +1,27 @@
+const urlParser = require('../src/backend/utils/url-parser');
+
+describe('url-parser tests', () => {
+  const testUrlWithPort = 'http://127.0.0.1:9200';
+  const testUrlWithoutPort = 'http://127.0.0.1';
+  const testport = '9200';
+  const expectedReturn = 'http://127.0.0.1:9200/';
+  test('urlParser with double ports', () => {
+    const result = urlParser(testUrlWithPort, testport);
+    expect(result).toBe(expectedReturn);
+  });
+
+  test('urlParser with 1 port', () => {
+    const result = urlParser(testUrlWithoutPort, testport);
+    expect(result).toBe(expectedReturn);
+  });
+
+  test('urlParser with invalid url', () => {
+    const result = urlParser('', testport);
+    expect(result).toBe(null);
+  });
+
+  test('urlParser with url with no port and no port', () => {
+    const result = urlParser(testUrlWithoutPort, '');
```

Comment:
Add test cases for:

- port is undefined
- port is null
- port is a number
- port is a string
- port is an invalid string
- port is a number outside the port range (e.g., 65535 + 1)